pkgs/applications/misc/cipher/default.nix

```
@@ -0,0 +1,56 @@
+{ stdenv
+, fetchFromGitHub
+, meson
+, ninja
+, pkgconfig
+, pantheon
+, python3
+, gettext
+, glib
+, gtk3
+, hicolor-icon-theme
+, libgee
+, wrapGAppsHook }:
+
+stdenv.mkDerivation rec {
+  pname = "cipher";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "arshubham";
+    repo = "cipher";
+    rev = version;
+    sha256 = "0n5aigcyxnl4k52mdmavbxx6afc1ixymn3k3l2ryhyzi5q31x0x3";
+  };
+
+  nativeBuildInputs = [
+    gettext
+    meson
+    ninja
+    pantheon.vala
+    pkgconfig
+    python3
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    glib
+    gtk3
+    hicolor-icon-theme
+    pantheon.granite
+    libgee
+  ];
+
+  postPatch = ''
+    chmod +x ./post_install.py
+    patchShebangs ./post_install.py
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A simple application for encoding and decoding text";
+    homepage = "https://github.com/arshubham/cipher";
+    maintainers = with maintainers; [ kjuvi ] ++ pantheon.maintainers;
+    platforms = platforms.linux;
+    license = licenses.gpl3;
```

```suggestion
    license = licenses.gpl3Plus;
```

https://github.com/arshubham/cipher/blob/master/src/Application.vala#L7